src/kinetics/Reaction.cpp

```diff
@@ -253,11 +308,129 @@ std::string Reaction::equation() const
 void Reaction::setEquation(const std::string& equation, const Kinetics* kin)
 {
     parseReactionEquation(*this, equation, input, kin);
+
+    std::string rate_type = input.getString("type", "");
+    if (rate_type == "three-body") {
+        // state type when serializing
+        m_explicit_rate = true;
+    } else if (rate_type == "elementary") {
+        // user override
+        m_explicit_rate = true;
+        return;
+    } else if (rate_type == "two-temperature-plasma") {
+        // two-temperature-plasma reactions do not use third bodies
+        return;
+    } else if (kin && kin->thermo(kin->reactionPhaseIndex()).nDim() != 3) {
+        // interface reactions
+        return;
+    }
+
+    std::string third_body;
+    size_t count = 0;
+    size_t countM = 0;
+    for (const auto& reac : reactants) {
+        // detect explicitly specified collision partner
+        if (products.count(reac.first)) {
+            third_body = reac.first;
+            size_t generic = third_body == "M" || ba::ends_with(third_body, "M)");
```

Comment:
Isn't just looking for names ending with `M` likely to generate some false positives?